library/Core/src/Juvix/Core/Erased/Ann/Conversion.hs

```diff
@@ -182,13 +192,21 @@ toRaw t@(ErasedAnn.Ann {term}) = t {ErasedAnn.term = toRaw1 term}
     argsToTerms ts xs = go (toList ts) xs
       where
         go _ [] = []
-        go (_ : ts) (App.TermArg a : as) =
-          takeToTerm a : go ts as
+        go (_ : ts) (App.TermArg (App.Return ty term) : as) =
+          returnToTerm (App.Return ty term) : go ts as
         go (t : ts) (App.VarArg x : as) =
           varTerm t x : go ts as
         go [] (_ : _) =
           -- a well typed application can't have more arguments than arrows
           undefined
+        returnToTerm (App.Return ty term) =
+          ErasedAnn.Ann
+            { usage = Usage.Omega, -- FIXME should usages even exist after erasure?
+              type' = ErasedAnn.fromPrimType ty,
+              term = ErasedAnn.Prim term
+            }
+        returnToTerm (App.Cont take args nat) = notImplemented
+
```

Comment:
This function shows up again. 

We should unify this with the other one, as it's hard to maintain copies.

Also in this case, could we pass in an extra usage like we did with the other one, or do we not have that in scope anywhere?